test/numerical/CMakeLists.txt

```diff
@@ -53,11 +53,11 @@ endfunction()
 # of the flags for exceptions (among others) are not set correctly.
 llvm_update_compile_flags(rapidcheck)
 
-set(TEST_LINK_LIBS rapidcheck CompilerUtils ExecutionSession)
+set(TEST_LINK_LIBS rapidcheck CompilerUtils ExecutionSession ModelLib)
```

Comment:
Since ModelLib requires ExecutionSession and CompilerUtils, you can drop them from the list.

Reply:
sure, did it, but is there any issues with having redundant dependences, to make is super clear?

Reply:
Had to reintroduce it because it was causing errors:

```
In file included from /workdir/onnx-mlir/test/numerical/TestMatMul2D.cpp:15:
/workdir/onnx-mlir/src/Compiler/CompilerUtils.hpp:29:10: fatal error: onnx-mlir/Compiler/OMCompilerTypes.h: No such file or directory
   29 | #include "onnx-mlir/Compiler/OMCompilerTypes.h"
```

Reply:
There's generally no issue with redundant dependencies -- but by avoiding it, we make sure that going forward we don't end up including unneeded libs as dependencies which slow down builds and increase the size of the produced binaries.

It looks like you marked `CompilerUtils` and `ExecutionSession` as `private` libs for the linking of `ModelLib`. It would be better if you marked them as `public` instead and removed them form here.